server/routes/ssotRoutes.test.js

```diff
@@ -362,3 +362,126 @@ describe('ssot/getAllAttributes/:robotId', () => {
+  });
+
+  it('successfully deletes the user access object to a robot', async () => {
+    await dbLoader.loadSsotInDb();
+    await dbLoader.loadUserAccessObjectsInDb();
+
+    const loadedUserAccessObjects = await mongoose
+      .model('userAccessObject')
+      .find()
+      .exec();
+    expect(loadedUserAccessObjects.length).toBe(2);
+
+    const request = httpMocks.createRequest({
+      method: 'DELETE',
+      params: {
+        robotId: testRobotId,
+      },
+    });
+    const response = httpMocks.createResponse();
+
+    await ssotRetrievalController.deleteRobot(request, response);
+    expect(response.statusCode).toBe(200);
+
+    // verify if really deleted
+    const foundUserAccessObjects = await mongoose
+      .model('userAccessObject')
+      .find()
+      .exec();
+    expect(foundUserAccessObjects.length).toBe(
+      loadedUserAccessObjects.length - 1
+    );
+
+    const usableTestRobotId = mongoose.Types.ObjectId(testRobotId);
+    const foundUserAccessObjectsById = await mongoose
+      .model('userAccessObject')
+      .find({ robotId: usableTestRobotId })
+      .exec();
+
+    expect(foundUserAccessObjectsById.length).toBe(0);
+  });
+
+  it('successfully deletes the attributes to a robots activities', async () => {
+    await dbLoader.loadSsotInDb();
+    await dbLoader.loadAttributesInDb();
+
+    const loadedAttributes = await mongoose
+      .model('rpaAttributes')
+      .find()
+      .exec();
+    expect(loadedAttributes.length).toBe(3);
+
+    const request = httpMocks.createRequest({
+      method: 'DELETE',
+      params: {
+        robotId: testRobotId,
+      },
+    });
+    const response = httpMocks.createResponse();
+
+    await ssotRetrievalController.deleteRobot(request, response);
+    expect(response.statusCode).toBe(200);
+
+    // verify if really deleted
+    const foundAttributes = await mongoose.model('rpaAttributes').find().exec();
+    expect(foundAttributes.length).toBe(0);
+    expect(foundAttributes.length).not.toBe(loadedAttributes.length);
+    expect(foundAttributes).not.toBe(loadedAttributes);
+  });
+
+  it('successfully deletes the parameters to a robots activities', async () => {
+    await dbLoader.loadSsotInDb();
+    await dbLoader.loadParametersInDb();
+
+    const loadedParameters = await mongoose.model('parameter').find().exec();
+    expect(loadedParameters.length).toBe(3);
+
+    const request = httpMocks.createRequest({
+      method: 'DELETE',
+      params: {
+        robotId: testRobotId,
+      },
+    });
+    const response = httpMocks.createResponse();
+
+    await ssotRetrievalController.deleteRobot(request, response);
+    expect(response.statusCode).toBe(200);
+
+    // verify if really deleted
+    const foundParameters = await mongoose.model('parameter').find().exec();
+    expect(foundParameters.length).toBe(0);
+    expect(foundParameters.length).not.toBe(loadedParameters.length);
+    expect(foundParameters).not.toBe(loadedParameters);
+  });
```

Comment:
Please also add a test for the jobs once implement and if you could also a job, where everything is deleted at once (attributes, parameter, jobs)

Reply:
Added a job test, although I do not understand the usefulness of another completely covering test. To my understanding, if all individual tests pass, the complete coverage is given. What makes you think otherwise?

Reply:
Thanks again here for the two additional tests 👍